contracts/oracle/PCVOracle.sol

```diff
@@ -0,0 +1,388 @@
+
+    /// @notice return all addresses listed as illiquid venues
+    function getIlliquidVenues() external view returns (address[] memory) {
+        return illiquidVenues.values();
+    }
+
+    /// @notice return all addresses that are liquid or illiquid venues
+    function getAllVenues() external view returns (address[] memory) {
+        uint256 liquidVenueLength = liquidVenues.length();
+        uint256 illiquidVenueLength = illiquidVenues.length();
+        address[] memory allVenues = new address[](
+            liquidVenueLength + illiquidVenueLength
+        );
+        uint256 j = 0;
+
+        /// there will never be more than 100 total venues
+        /// so keep the math unchecked to save on gas
+        unchecked {
+            for (uint256 i = 0; i < liquidVenueLength; i++) {
+                allVenues[j] = liquidVenues.at(i);
+                j++;
+            }
+
+            for (uint256 i = 0; i < illiquidVenueLength; i++) {
+                allVenues[j] = illiquidVenues.at(i);
+                j++;
+            }
+        }
+
+        return allVenues;
+    }
+
+    /// @return the ratio of liquid to illiquid assets in the Volt system
+    /// using stale values and not factoring any interest or losses sustained
+    /// but not realized within the system
+    /// value is scaled up by 18 decimal places
+    function getLiquidVenuePercentage() public view returns (uint256) {
+        return
+            (SCALE * lastLiquidBalance) /
+            (lastIlliquidBalance + lastLiquidBalance);
+    }
+
+    /// @notice check if a venue is in the list of illiquid venues
+    /// @param illiquidVenue address to check
+    /// @return boolean whether or not the illiquidVenue is in the illiquid venue list
+    function isIlliquidVenue(address illiquidVenue)
+        external
+        view
+        returns (bool)
+    {
+        return illiquidVenues.contains(illiquidVenue);
+    }
+
+    /// @notice check if a venue is in the list of illiquid venues
+    /// @param liquidVenue address to check
+    /// @return boolean whether or not the liquidVenue is in the illiquid venue list
+    function isLiquidVenue(address liquidVenue) external view returns (bool) {
+        return liquidVenues.contains(liquidVenue);
+    }
+
+    /// @notice check if a venue is in the list of liquid or illiquid venues
+    /// @param venue address to check
+    /// @return boolean whether or not the venue is part of the liquid or illiquid venue list
+    function isVenue(address venue) external view returns (bool) {
+        return liquidVenues.contains(venue) || illiquidVenues.contains(venue);
+    }
+
+    /// @notice get the total PCV balance by looping through the liquid and illiquid pcv deposits
+    /// @dev this function is meant to be used offchain, as it is pretty gas expensive.
+    function getTotalPcv()
+        external
+        view
+        returns (
+            uint256 liquidPcv,
+            uint256 illiquidPcv,
+            uint256 totalPcv
```

Comment:
todo: return `unclaimedLiquidPcvYield` (diff between `liquidPcv` that we read live, and the stored `lastLiquidBalance`) + do the same for illiquid

Reply:
not very useful in retrospect